proposals/0012-f3-deals-reliability-ux.md

```diff
@@ -0,0 +1,159 @@
+
+#### Confidence
+
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+3. We have very solid evidence that the deals flow is... suboptimal. We've used this input to design our action plan. Note that this doc is just a draft and indexes mostly on the problem and not so much in the solution. We'll discover more action items as we execute on the project, but we're confident in the trajectory and themes we're setting.
+
+## Project definition
+
+#### Brief plan of attack
+
+**UX track**
+
+1. Study the user flow in detail. Inventorise the specific elements that are confusing users.
+2. Initiate discussions about user-friendlier terminology.
+3. Design a new CLI UX, prototype it, iterate.
+4. Implement CLI UX changes (and underlying code changes)
+5. One-stop set of JSON-RPC operations that encapsulate the new behaviour with no frills (without leaking implementation details as much as possible).
+
+**Technical reliability track**
+
+0. Do a full inventory of outstanding issues; get to the bottom of the long-standing mystical issues; analyse; fix. Some known tasks:
+1. Testing: Design and implement test suites to reproduce the concrete issues and serve as regression tests. Nightly tests for deals.
+2. Fix memory footprint issues.
+3. Fix resumability of data transfers.
+4. Consider p2p file-oriented transfer protocol to cut out complexity for now.
+5. Identify areas of the deals flow code that need more informative logging (eg data transfer restarts / connection retries).
+6. Add logs and journal events.
+7. Fix miner datastore bloat.
+
+#### What does done look like?
+
+_What specific deliverables should completed to consider this project done?_
+
+1. A new version of Lotus (and associated modules) that achieves the success criteria.
+2. A new interactive CLI command (`lotus store`) that guides the user through the process of performing a storage deal, and provides visibility and a polished flow.
+3. A simplified, porcelain S3-like JSON-RPC API that covers the deals flow and provides the logical sequence of steps outlined in previous sections, under a refined, well-documented, and opaque API that doesn't leak Filecoin implementation details.
+4. (Optionally) A prototype for a deals dashboard web UI (in collaboration with The Spark).
+
+#### What does success look like?
+
+_Success means impact. How will we know we did the right thing?_
+
+- 80% of deals initiated via the CLI result in success, as long as prerequisites are met (enough balance, synced node, etc.) under a variety of conditions (e.g. connection losses, slow bandwidth, concurrent deals).
```

Comment:
Maybe: # of deals made on the network, # of clients who have made at least 1 successful deal, number of projects actively storing data on Filecoin

Reply:
Do we have metrics today on deals initiated via the CLI and then which portion of those are successful?  If not, should we add that to the project plan to first implement/release those metrics so we can see the impact of the F3 work?

Reply:
@momack2 how would you feel about dusting off the dealbot we used in Space Race, and "letting it loose" on mainnet?